src/lib/bootloader/proposal_client.rb

```diff
@@ -337,10 +343,28 @@ def handle_errors(ret)
 
     def single_click_action(option, value)
       stage1 = ::Bootloader::BootloaderFactory.current.stage1
-      devices = (option.to_sym == :mbr) ? stage1.boot_disk_names : stage1.boot_partition_names
+      case option
+      when "boot_mbr"
+        devices = stage1.boot_disk_names
+      when "boot_boot"
+        devices = stage1.boot_partition_names
+      when "trusted_boot"
+        ::Bootloader::BootloaderFactory.current.trusted_boot = value
+      when "secure_boot"
+        ::Bootloader::BootloaderFactory.current.secure_boot = value
+        if value && Yast::Arch.s390
+          Yast2::Popup.show(
+            _(
+              "Make sure to also enable Secure Boot in the HMC.\n\n" \
+              "Otherwise this system will not boot."
+            ),
+            headline: :warning, buttons: :ok
+          )
+        end
+      end
       log.info "single_click_action #{option} #{value.inspect} #{devices}"
 
-      devices.each do |device|
+      devices&.each do |device|
```

Comment:
this starting to feel a bit too hacky, as it is related only to "boot_" arguments. What about moving it inside case like:
```suggestion
      bootloader = ::Bootloader::BootloaderFactory.current
      case option
      when /boot_/
        stage1 = bootloader.stage1
        devices = option == "boot_mbr" ? stage1.boot_disk_names : stage1.boot_partition_names
        log.info "single_click_action #{option} #{value.inspect} #{devices}"
        devices.each do |device|
          value ? stage1.add_udev_device(device) : stage1.remove_device(device)
        end
      when "trusted_boot"
        bootloader.trusted_boot = value
      when "secure_boot"
        bootloader.secure_boot = value
      end
      
      Yast::Bootloader.proposed_cfg_changed = true
    end
```